app/models/patient.rb

```diff
@@ -474,81 +438,26 @@ def self_reporter_or_proxy?
     responder_id == id
   end
 
+  # Patient name to be displayed in linelist
+  def displayed_name
+    first_name.present? || last_name.present? ? "#{last_name}#{first_name.blank? ? '' : ', ' + first_name}" : 'NAME NOT PROVIDED'
+  end
+
   # Allow information on the monitoree's jurisdiction to be displayed
   def jurisdiction_path
     jurisdiction&.path&.map(&:name)
   end
 
   # Single place for calculating the end of monitoring date for this subject.
   def end_of_monitoring
+    return 'Continuous Exposure' if continuous_exposure
     return (last_date_of_exposure + ADMIN_OPTIONS['monitoring_period_days'].days)&.to_s if last_date_of_exposure.present?
     return (created_at + ADMIN_OPTIONS['monitoring_period_days'].days)&.to_s if created_at.present?
   end
 
-  # Is this patient symptomatic?
-  def symptomatic?
-    assessments.where(symptomatic: true).exists?
-  end
-
-  # Is this patient symptomatic?
-  def asymptomatic?
-    (!latest_assessment.nil? &&
-     assessments.where(symptomatic: true).count.zero? &&
-     latest_assessment.created_at >= ADMIN_OPTIONS['reporting_period_minutes'].minutes.ago) ||
-      (created_at && latest_assessment.nil? &&
-       created_at >= ADMIN_OPTIONS['reporting_period_minutes'].minutes.ago)
-  end
-
-  # Is this patient non_reporting?
-  def non_reporting?
-    (!latest_assessment.nil? &&
-     assessments.where(symptomatic: true).count.zero? &&
-     latest_assessment.created_at < ADMIN_OPTIONS['reporting_period_minutes'].minutes.ago) ||
-      (latest_assessment.nil? && created_at && created_at < ADMIN_OPTIONS['reporting_period_minutes'].minutes.ago)
-  end
-
-  # Is this patient under investigation?
-  def pui?
-    monitoring && !purged && public_health_action != 'None'
-  end
-
-  # Has this patient purged?
-  def purged?
-    purged
-  end
-
-  # Has this patient purged?
-  def closed?
-    !monitoring && !purged
-  end
-
-  # Current patient status
-  def status
-    return :purged if purged?
-    return :closed if closed?
-
-    unless isolation
-      return :exposure_under_investigation if pui?
-      return :exposure_symptomatic if symptomatic?
-      return :exposure_asymptomatic if asymptomatic?
-
-      return :exposure_non_reporting
-    end
-    return :isolation_asymp_non_test_based if Patient.where(id: id).isolation_asymp_non_test_based.exists?
-    return :isolation_symp_non_test_based if Patient.where(id: id).isolation_symp_non_test_based.exists?
-    return :isolation_test_based if Patient.where(id: id).isolation_test_based.exists?
-    return :isolation_reporting if Patient.where(id: id).isolation_reporting.exists?
-
-    :isolation_non_reporting
-  end
-
-  # Updated symptom onset date IF updated assessment happens to be the oldest symptomatic
-  def refresh_symptom_onset(assessment_id)
```

Comment:
Can you explain why this could be deleted? I see it's uses are deleted elsewhere but I'm not 100% on why.

Reply:
sure thing, so the purpose of this method before was to update the symptom_onset field whenever an assessment is created, which is not only its own field, but is also used to determine a patient's status in isolation (requiring review, non reporting, reporting, etc...).

this method was called in several places from the assessment controller (basically in places where assessments are created) and I actually found a bug along the way in that the symptom onset date was not being updated when assessments were updated.

for example, let's say a monitoree completed a symptomatic assessment and then an epi marked it as reviewed or edited it to be asymptomatic. the symptom onset date should be reset to `nil` because the symptom onset date is defined as the date of the first symptomatic assessment (which there are no longer any of).

part of the db changes of this PR include additional line list fields to help determine patient statuses faster and those fields that were added are similar to `symptom_onset` in that they are to be updated whenever an assessment is created/updated/deleted. the way those fields are updated is via active record callbacks, some of which are defined here: https://github.com/SaraAlert/SaraAlert/blob/SARAALERT-483/app/models/assessment.rb#L19

these callbacks methods are automatically called whenever an assessment is created/updated/deleted (except for certain bulk CRUD methods which we have to be careful of), and this is a better way of ensuring that these fields are updated at the proper times.

there are also additional callbacks for linelist fields related to symptoms, laboratories, and transfers in their respective models. let me know if any of this doesn't make sense or if you have other questions!

Reply:
That clarifies a lot, thank you!

Reply:
no problem!